### ISSUE TITLE
 Badly corrects the code.

### ISSUE THREAD

**Issue by themafia98** [prettier/prettier#5751]
Before:

```
<div
      class="main__image-list main-image-list--margin main__image-list-one">
</div>
```

After alt+shift+ctrl+P

```
<div
      class="main__image-list main-image-list--margin main__image-list-one"
    ></div>
```

Another example
Before:

```
   <link
      rel="stylesheet"href="https://use.fontawesome.com/releases/v5.6.3/css/all.css"  integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous" />
```

After:

```
   <link
      rel="stylesheet"
      href="https://use.fontawesome.com/releases/v5.6.3/css/all.css"
      integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/"
      crossorigin="anonymous"
    />
```

And more others tags (img.....)

this is a bug or a trick?

**Comment by lydell** [comment]
The `<link>` case is intended. There are several long attributes and they don’t fit on a single line, so Prettier put them on separate lines.

The `<div>` case looks odd though. If there’s just one attribute we don’t gain much from moving it to a new line.

**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4AIwA2AhgM4kC8AOiALZGZQD6jmdA5nALQGYky50GnVkQ7deMTpzoAnNgwH0mLdlx59O0OLgBmDIgU6FSJagD5UAeizYzIADQgIABxiZoJZKCIyZEAO4ACj4InihE2BCY6A4gAEYyRGAA1nAwAMrOSQxsyDAyAK5wjgAWMDQEAOolmPAkWWBw6aG1OLUAnsjgZLEMJHAyMIGJbHTIOgb9jgBWJAAeAEKJKWnpRDRwADIMcOOTxSCzc+k5BHAAigUQ8HsEUyBZMv0yXWUVsc4yDDCV0TAlyAAHAAGRyfCD9SqJZxdT5wZ7YXaOGRwACOBUwKOGojGSAmdwO-RomDyhUJpwuVxueP2jhgRDiv3Q-2QACY6YlMDwoGwAMIQGi4kBQLSxAr9AAqDLC+P6AF85UA)
```sh
--parser html
```

**Input:**
```html
<div class="main__image-list main-image-list--margin main__image-list-one final-class"></div>
```

**Output:**
```html
<div
  class="main__image-list main-image-list--margin main__image-list-one final-class"
></div>

```

Compare with JSX:

**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4AIwA2AhgM4kC8AOiALZGZQD6jmdA5nALQGYky50GnVkQ7deMTpzoAnNgwH0mLdlx59O0OLgBmDIgU6FSJagD5UAeizYzIADQgIABxiZoJZKCIyZEAO4ACj4InihE2BCY6A4gAEYyRGAA1nAwAMrOSQxsyDAyAK5wjgAWMDQEAOolmPAkWWBw6aG1OLUAnsjgZLEMJHAyMIGJbHTIOgb9jgBWJAAeAEKJKWnpRDRwADIMcOOTxSCzc+k5BHAAigUQ8HsEUyBZMv0yXXFEce0E0LHOMgwwlWiMBKyAAHAAGRy-CD9SqJZxdX5wZ7YXaOGRwACOBUwGOGojGSAmdwO-RomDyhVJpwuVxuRP2jhg70B6GByAATEzEpgeFA2ABhCA0QkgKBaWIFfoAFXeYWJ-QAvoqgA)
```sh
--parser babylon
```

**Input:**
```jsx
<div class="main__image-list main-image-list--margin main__image-list-one final-class"></div>
```

**Output:**
```jsx
<div class="main__image-list main-image-list--margin main__image-list-one final-class" />;

```

Maybe we should bring that special case to HTML as well.

**Comment by ikatyang** [comment]
The issue here is that we don't know if the content of the attribute can be broken into multiple line so we did not force one attribute to be single line, for example we may add support for `class` (#5482), which should break it into multiple line if it does not fit the print width.

**Comment by duailibe** [comment]
@ikatyang None of the examples have multiline attributes though.

I think what @lydell is suggesting is that we keep tags with a single attribute and no children in a single line even if it exceeds print width:

```html
<div class="main__image-list main-image-list--margin main__image-list-one final-class"></div>
```

**Comment by ikatyang** [comment]
I meant if we force 1 attr to be single line, the `breakParent` magic would no longer work:

**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAlgWwOYAIDOATmPnDALwA6IA9AIb6kz40Bu6+ArnQDYACABgB0AViEAHKHlwAWAQIDuAGlz1GZFuy68JU3PoMAOASOWUYappo7d+AJgAeu6bgCMANleKl5yxrY2vHwAzE6SLnYyrqbUqgB8IEogEOIw6ND4yKB0hIQQCgAKOQiZKHSsEOgAJokgAEaEdGAA1mQAyuJN6FLIMISccEkAFjCYPADqQ+jw+J1gcG0l0+jsMACeyOCMtd2khDAFjdiYdMgAZrykSQBW+A4AQo0t7XSYcAAy3XDnl4Mgtw42t1sDw4ABFTgQeA-HhXECdQh7TYjMa1cSEbowcbVGBDZDGJLoiCkcaNcSbdFwPasb5JQhwACOnHQ9MOdGOpyQF1hf1ImHQvX6vOBoIhUO+XN+SRgdDq2KquOQdmljXQPGBAGEIJgTpsoNBaSBOKQACqy0rc0gAXytQA)
```sh
--parser html
```

**Input:**
```html
<img srcset="/assets/visual@0.5.png  400w, /assets/visual.png      805w,	/assets/visual@2x.png   1610w,	/assets/visual@3x.png   2415w" />
```

**Output:**
```html
<img
  srcset="
    /assets/visual@0.5.png  400w,
    /assets/visual.png      805w,
    /assets/visual@2x.png  1610w,
    /assets/visual@3x.png  2415w
  "
/>

```

Output if we force 1 attr to be single line:

```html
<img srcset="
    /assets/visual@0.5.png  400w,
    /assets/visual.png      805w,
    /assets/visual@2x.png  1610w,
    /assets/visual@3x.png  2415w
  "/>
```

**Comment by duailibe** [comment]
Yeah obviously in that case we wouldn't force it to be single line, just like we don't force it with JSX:

**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAYhCACARgQwCcBeAHRADMtSpSYDDzsB6APgG4QAaECABxgBLaAGdkoIoQgB3AApEEYlPgBuEQQBNuIXIXxgA1nBgBlPvsFQA5shiEArnB4ALGAFsANgHVng+CPMwOBNFP0EVPwBPZHARMR5LEThCGFk9Kzd8ZAp8DySeACsRAA8AIT1DYxN8NzgAGUs4bNz8kCLik0srDzgARXsIeGa8pxBzQiTCGIJcSI9obT5CSxgvTRhnZAAOAAYeJYgkrz0+GKW4SZUmnkI4AEd7QVu0-AyspByRniS3QVsHUYiLo9fqDJofFqjGD4XBrDQbZAAJh4dnwgg8XQAwhA3JkYlBoNcQPYkgAVGFKT5JAC+1KAA)
```sh
--parser babylon
```

**Input:**
```jsx
<Foo bar="foo
	bar" />;
```

**Output:**
```jsx
<Foo
  bar="foo
	bar"
/>;

```

**Comment by mpro7** [comment]
Not sure if my case refers to this issue, but it seems kinda similar. I've set `html.format.wrapLineLength` and `prettier.printWidth` properties to the same value. Every time the line length is larger that defined, it causes not needed formatting - each property goes into the new line:
```
<grid-column
	field="formatCode"
	title="Format"
	width="65"
	class="status-column cell-with-button"
	*ngIf="width > 1"
	[class]="{ codeColumn: true }"
	[hidden]="columns[7].hidden"
></grid-column>
```
I would like to have the line wrapped only to necessary number of lines, which means, the next property which is exceed the line number goes to another line etc.:

```
<grid-column field="incoTermCode" title="Incoterm" width="65"
	class="status-column cell-with-button" *ngIf="width > 1"
	[class]="{ codeColumn: true }" [hidden]="columns[7].hidden"
></grid-column>
```
I'm not sure if this feature already exists, but it would be nice to have it.

**Comment by j-f1** [comment]
That’s not the way Prettier works. We decided it’s usually more readable to split each thing onto its own line.

**Comment by mpro7** [comment]
Could be this feature possible as optional? If not, could somebody suggest please another tool/solution to get expected result?

**Comment by duailibe** [comment]
@mpro7 No, it's been suggested in the past and we've already ruled out. An alternative is to fork Prettier and change this behavior.

**Comment by mpro7** [comment]
@duailibe thanks for your suggestion. I may have a try, will be this possible to release with prettier or as forked side-project only?

**Comment by duailibe** [comment]
As forked side-project only.

**Comment by mpro7** [comment]
@duailibe pretty sad, so maybe there is an option to disable just Prettier line wrapping - I cannot see it right now? I do not want to remove Prettier at all, because I like it's other features, but I think that with this option I would get what I want with inbuilt Visual Studio Code formatting rules.

**Comment by lydell** [comment]
Correct. Line wrapping is at the heart of Prettier and cannot be disabled.

**Comment by sjMonsta** [comment]
I had the same issue and realized @lydell is right. So I went op my settings, searched prettier and found 'Prettier: Print Width' option which was set to 80. I increased it to 150 and it worked for me.

**Comment by andrew-ma** [comment]
Thanks sjMonsta, it works.
"prettier.printWidth": 150

**Comment by mhamri** [comment]
+1, breaking 100 attribute to 100 line doesn't make sense. the `printWidth` rule should be respected and when there is still room to fill in one line, why break things in a new line? 

![image](https://user-images.githubusercontent.com/8991783/72036960-7590c200-32d7-11ea-87f0-61b92921eff2.png)


here for a small `icon=something` & size="m" spared one line each?

**Comment by silvermario** [comment]
> why break things in a new line?

I would say fewer merge conflicts and cleaner change history.
Though i would love if there was some threshold and until reaching it prettier wouldn't break attributes into multiline format.

**Comment by amcolash** [comment]
This is exactly what `printWidth` (default: 80) does @silvermario, per the docs:
```
Print Width
Specify the line length that the printer will wrap on.
```

**Comment by mhamri** [comment]
@silvermario it should be a separate option, `html.respectLineBreak:false` 
if not the line break is a rule, it shouldn't behave two different way for two different file type

**Comment by Lonli-Lokli** [comment]
I cannot understand the reason for putting every attribute on separate line.
With the new version I see changes which will be valid for this case as well

```
// Prettier 1.19
if (
  foo
    .one()
    .two()
    .three() ===
  bar
    .four()
    .five()
    .six()
) {
  // ...
}

// Prettier 2.0
if (foo.one().two().three() === bar.four().five().six()) {
  // ...
}
```

**Comment by lydell** [comment]
@Lonli-Lokli Hi! Have you read through the thread? As mentioned, if the attributes don’t fit in a single line they’re split into one per line. That’s how Prettier works. If you don’t like that, Prettier is probably not the tool you are looking for.

**Comment by Lonli-Lokli** [comment]
@lydell my concern is not about splitting single attribute into several lines but splitting each attribute on separate line, like mentioned in https://github.com/prettier/prettier/issues/5751#issuecomment-454776010.

That's the way how Js works before 2.0

**Comment by lydell** [comment]
@Lonli-Lokli I’m sorry, I don’t understand what you mean.

**Comment by warcraft14115** [comment]
[https://github.com/prettier/prettier/issues/5751#issuecomment-454776010](https://github.com/prettier/prettier/issues/5751#issuecomment-454776010)

This will work. Go to setting and find html.format.wrapLineLength and prettier.printWidth change to  300 :) Work perfect. 
Ex:
```
<input 
className="input disabled" 
type="number" 
placeholder="Origin price " 
value="59" />
```

will change to  `<input className="input disabled" type="number" placeholder="Origin price " value="59" />`
No hurt to your eyes

**Comment by potasium56** [comment]
Is there a way i can make self-closing HTML elements with many attributes not make then each on one line its annoying and looks bad to me its not prettier XD. I would rather have it on one line

**Comment by tahonaPL** [comment]
problem still exist does someone has solution for this ?

**Comment by EvelRus** [comment]
up = )

**Comment by liudonghua123** [comment]
It's almost 2.5 years. 😢

**Comment by chen-ccy** [comment]
> 不确定我的案例是否涉及这个问题，但看起来有点相似。我已将[Prettier VSC 扩展](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) `html.format.wrapLineLength`和`prettier.printWidth`属性设置为相同的值。每次行长大于定义的长度时，都会导致不需要的格式 - 每个属性都进入新行：
> 
> ```
> <grid-column
> 	field="formatCode"
> 	title="Format"
> 	width="65"
> 	class="status-column cell-with-button"
> 	*ngIf="width > 1"
> 	[class]="{ codeColumn: true }"
> 	[hidden]="columns[7].hidden"
> ></grid-column>
> ```
> 
> 我希望只将行换行到必要的行数，这意味着超过行长度的下一个属性转到下一行等：
> 
> ```
> <grid-column field="incoTermCode" title="Incoterm" width="65"
> 	class="status-column cell-with-button" *ngIf="width > 1"
> 	[class]="{ codeColumn: true }" [hidden]="columns[7].hidden"
> ></grid-column>
> ```
> 
> 我不确定这个功能是否已经存在，但拥有它会很好。

I think you're right, when there are many properties, it will take up a lot of rows. Even each item of the array wraps. I think it makes sense to wrap when the property doesn't fit in one line

**Comment by rp-feflo** [comment]
It would be really nice to avoid line break of elements with a single attribute like @lydell mentioned above.
Setting `printWidth` to a larger number avoids all elements from being separated into new lines. But you can only choose between true or false. The case is to only break attributes into an new line if there are more than one attribute set to an tag or configure the number of attributes you want to stay in one line:

`
<div id="my-awesome-id"></div>

<div id="my-awesome-id">
    ...
</div>
`

vs.

`
<div 
    id="my-awesome-id"
    class="class-one class-two"
>
</div>

<div 
    id="my-awesome-id"
    class="class-one class-two"
>
    ...
</div>
`

It wold be awesome to set something like "attributesPerLine": 3

`
<a href="#" title="..." id="my-awesome-id">
    ...
</a>

<a 
    href="#" 
    title="..." 
    id="my-awesome-id" 
    class="class-one class-two"
>
    ...
</a>
`
